sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   A pool of <see cref="TransportProducer" />.
+    ///   It periodically checks if the producers are active and if not it disposes them.
+    /// </summary>
+    ///
+    internal class TransportProducerPool : IAsyncDisposable
+    {
+        /// <summary>The period in minutes after which <see cref="PerformExpiration"/> is run</summary>
+        private const int PerformExpirationPeriodAsMinutes = 10;
+
+        /// <summary>
+        ///   The set of active Event Hub transport-specific producers specific to a given partition;
+        ///   intended to perform delegated operations.
+        /// </summary>
+        ///
+        private ConcurrentDictionary<string, PoolItem> Pool { get; set; } = new ConcurrentDictionary<string, PoolItem>();
```

Comment:
nit:  Would we need/want to `set` this?

Reply:
I needed this for the tests, however I realised that it might be just ok adding it as a defaulted parameter in the constructor?

Reply:
I think the change that you've got where you're allowing it to be injected as part of the internal constructor is a good approach!